docs/src/main/sphinx/security/inspect-jks.rst

```diff
@@ -0,0 +1,128 @@
+
+  Example:
+
+  .. code-block:: text
+
+    SubjectAlternativeName [
+        DNSName:  cluster.example.com
+    ]
+
+If your keystore shows valid information for your cluster, proceed to configure
+the Trino server, as described in :ref:`cert-placement` and
+:ref:`configure-https`.
+
+The rest of this page describes additional steps that may apply in certain
+circumstances.
+
+.. _import_to_keystore:
+
+Extra: add PEM to keystore
+--------------------------
+
+Your site may have standardized on using JKS semantics for all servers. If a
+vendor sends you a PEM-encoded certificate file for your Trino server, you can
+import it into a keystore with a command like the following. Consult ``keytool``
+references for different options.
+
+.. code-block:: shell
+
+  keytool -trustcacerts -import -alias cluster -file localhost.pem -keystore localkeys.jks
+
+If the specified keystore file exists, ``keytool`` prompts for its password. If
+you are creating a new keystore, ``keytool`` prompts for a new password, then
+prompts you to confirm the same password. ``keytool`` shows you the
+contents of the key being added, similar to the ``keytool -list`` format, then
+prompts:
+
+.. code-block:: text
+
+  Trust this certificate? [no]:
+
+Type ``yes`` to add the PEM certificate to the keystore.
+
+The ``alias`` name is an arbitrary string used as a handle for the certificate
+you are adding. A keystore can contain multiple keys and certs, so ``keytool``
+uses the alias to address individual entries.
+
+.. _cli_java_truststore:
+
+Extra: Java truststores
+-----------------------
+
+.. note::
+
+    Remember that there may be no need to identify a local truststore when
+    directly using a signed PEM-encoded certificate, independent of a keystore.
+    PEM certs can contain the server's private key and the certificate chain all
+    the way back to a recognzied CA.
+
+Truststore files contain a list of :ref:`Certificate Authorities <glossCA>`
+trusted by Java to validate the private keys of servers, plus a list of the
+certificates of trusted TLS servers. The standard Java-provided truststore file,
+``cacerts``, is part of your Java installation in a standard location.
+
+Keystores normally rely on the default location of the system truststore, which
+therefore does not need to be configured.
+
+However, there are cases in which you need to use an alternate truststore. For
+example, if your site relies on the JKS system, your network managers may have
+appended site-specific, local CAs to the standard list, to validate locally
+signed keys.
+
+If your server must use a custom truststore, identify its location in the
+server's config properties file. For example:
+
+.. code-block:: text
+
+   http-server.https.truststore.path=/mnt/shared/certs/localcacerts
+   http-server.https.truststore.key=<truststore-password>
+
+If connecting clients such as browsers or the Trino CLI must be separately
+configured, contact your site's network administrators for assistance.
```

Comment:
In a future update we should document how to deal with this in the CLI docs and the JDBC driver docs for starters.. nothing to do for now.

Reply:
Some of how-to-make-your-clients conform is in upcoming work already.

Reply:
Future